tests/test_sql_query.py

```diff
@@ -871,6 +871,53 @@ async def test_sql_query_44(self):
         ):
             await self.squery_values('SELECT name FROM User')
 
+    async def test_sql_query_45(self):
+        with self.assertRaisesRegex(
+            asyncpg.InvalidColumnReferenceError,
+            'duplicate column name: `a`',
+            position="16",
+        ):
+            await self.squery_values('SELECT 1 AS a, 2 AS a')
+
+    async def test_sql_query_46(self):
+        res = await self.scon.fetch(
+            '''
+            WITH
+              x(a) AS (VALUES (1)),
+              y(a) AS (VALUES (2)),
+              z(a) AS (VALUES (3))
+            SELECT * FROM x, y JOIN z u on TRUE
+            '''
+        )
+
+        # `a` would be duplicated,
+        # so second and third instance are prefixed with rel var name
+        self.assert_shape(res, 1, ['a', 'ya', 'ua'])
+
+    async def test_sql_query_47(self):
+        res = await self.scon.fetch(
+            '''
+            WITH
+              x(a) AS (VALUES (1)),
+              y(a) AS (VALUES (2), (3))
+            SELECT x.*, u.* FROM x, y as u
+            '''
+        )
+        self.assert_shape(res, 2, ['a', 'ua'])
+
+    async def test_sql_query_48(self):
+        res = await self.scon.fetch(
+            '''
+            WITH
+              x(a) AS (VALUES (1)),
+              y(a) AS (VALUES (2), (3))
+            SELECT * FROM x, y, y
+            '''
+        )
+
+        # duplicate rel var names can yield duplicate column names
+        self.assert_shape(res, 4, ['a', 'ya', 'ya'])
```

Comment:
We probably don't want this behavior, but I'm not sure what we'd want instead.

Error out? This might break some existing SQL tools / user queries.

Emit `y1a`, `y2a` instead?

Reply:
It's unlikely that tooling relies on colnames in those cases as opposed to col indexes, so emit fully-disambiguated names always.

Reply:
If we error out, that might break tools that were looking into results based on index.